src/hooks/useDetails.tsx

```diff
@@ -1,13 +1,23 @@
 import {useCallback, useEffect, useState, useRef} from 'react'
 
-function useDetails({ref, closeOnOutsideClick, defaultOpen, onClickOutside} = {}) {
+type useDetailsParameters = {
+  ref?: {current: HTMLElement}
```

Comment:
I'm wondering if there is an existing type for refs that I can import or if it's just best to do it inline like I have it here?

Reply:
This should do the trick :)

```suggestion
  ref?: React.RefObject<HTMLElement>
```